proposals/0006-gep.md

```diff
@@ -0,0 +1,99 @@
+---
+MEP: 0006
+Title: Chat Streaming Support
+Discussion:
+Implementation:
+---
+
+# Chat Streaming Support
+
+## Abstract
+
+Glide will support chat streaming endpoints as this is important feature with AI Chatbots. Additionally, this helps with latency issues seen with non-streaming responses.
+
+## Motivation
+
+Streaming of LLM responses is an important UX feature for AI chatbots. As chatbot use cases increase in complexity,
+response times from LLMs also increase. Streaming provide an enhanced UX as the end user actively sees the LLM generating the response.
+Additionally, Glide has the intention to support other streaming uses cases such as speech-to-text and text-to-speech.
+These intial efforts will support additional streaming use cases.
+
+### Requirements
+
+- R1: Streaming should be supported via the standard `v1/language` endpoint.
+- R2: Streaming should be enabled by default.
+- R3: Streaming should be specified in the Config if non-default values are needed.
+- R4: Streaming should be handled at the router level. A router is either streaming or not streaming.
+- R5: Initially, failures should be handled via a retry of the entire input prompt/message.
```

Comment:
I thought about this after we had agreed to just retry on failures. I think that this may be valuable to send some special message to the end application to indicate that we are going to retry. Then, having that marker applications will be able to react somehow if applicable.

Without that, there is no way for the app to react to retry but they may have unfortunate side effects. For example, in the chat apps like OpenAI chat, they would have duplicated response message. If they had a way to know that the error happens, I as a app developer would clear the previously output text and start over which is a way better experience. 

So the long story short, with this approach, having a way to identify that the error has happened seem to be useful.